snapcraft/cli/lifecycle.py

```diff
@@ -201,11 +201,10 @@ def _run_pack(snap_command: List[str]) -> str:
 def _pack(
     directory: str, *, compression: Optional[str] = None, output: Optional[str]
 ) -> None:
-    snap_path = shutil.which("snap")
-    if snap_path is None:
-        raise errors.HostToolNotFoundError(command_name="snap", package_name="snapd")
 
-    command = [snap_path, "pack"]
+    snap_path = file_utils.get_host_tool_path(command_name="snap", package_name="snapd")
+
+    command = [str(snap_path), "pack"]
```

Comment:
I don't think str is not entirely necessary here, subprocess does the right thing

Reply:
Hrm, you're right indeed.  I don't see that in the docs anywhere but some googling shows that it should work, and where it doesn't work in subprocess has been treated like a bug.

Reply:
done